src/dpi.erl

```diff
@@ -85,6 +100,13 @@ load_unsafe() ->
 %   local helper functions
 %===============================================================================
 
+reg(SlaveNode, Pid) ->
+    Name = {?MODULE, SlaveNode, make_ref()},
+    case global:register_name(Name, Pid) of
+        yes -> SlaveNode;
+        no -> {error, "failed to register process globally"}
```

Comment:
I think if the reg fails then the Slave node have to be stopped, as from the caller point of view the load call failed.

Reply:
Done in `load/1` instead, as `{error, {already_running, ...}}` case from `load` should not stop the slave node

Reply:
Done in `load/1` instead, as `{error, {already_running, ...}}` case from `load` should not stop the slave node